### PR TITLE
Limit Usage, Self-Ping, and Misc FE changes

### LINKED ISSUE #217: new user gets error when creating new trip
1. create new user (noreen did this on prod)
2. Try creating new trip with 3 stages per day, budget 100, 8 days
3. Should incorrectly get an error that "trip must be more than 0 days"

If this is not reproduceable then just leave it LOL 
i might have messed up

### LINKED ISSUE #221: Limit Usage
- cookie
OR
- db

### PR DESCRIPTION
**Pull request by NLmeng**
closes #221 closes #217

### CODE REVIEW

File: server/config/rateLimiter.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable no-unused-vars */
+const rateLimit = require('express-rate-limit')
```
Comment (AndyLiang1) [inline server/config/rateLimiter.js]: Should we move this file to middleware? Looks like a middleware cus i saw this.router.post('', apiLimiter, this.create.bind(this))

File: client/src/SessionController.js [hunk 2]
```diff
@@ -116,70 +117,71 @@ export function SessionController({ children }) {
 
   useEffect(() => {
     if (userStates.status === REQUEST_STATE.LOGGINGIN) {
+      dispatch(fetchLimitLeftAsync())
```
Comment (Graywing13) [inline client/src/SessionController.js:120]: is this supposed to be assigned to smt? seems like `fetchLimitLeftAsync()` is supposed to return smt, but it's not assigned to any variable here or w/e

File: client/src/SessionController.js [hunk 3]
```diff
@@ -116,70 +117,71 @@ export function SessionController({ children }) {
...
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [modalStates, enqueueSnackbar])
 
   useEffect(() => {
-    if (alertOpen) {
-      const timer = setTimeout(() => {
-        setAlertOpen(false)
-      }, 5000)
-      return () => clearTimeout(timer)
+    if (userStates.status === REQUEST_STATE.LOGGEDIN) {
```
Comment (Graywing13) [inline client/src/SessionController.js:167]: opinion: i don't think the user needs to know how many trips they have left unless they're trying to edit smt. but if u want to leave this in, it's fine by me 🏕️

File: client/src/SessionController.js [hunk 4]
```diff
@@ -116,70 +117,71 @@ export function SessionController({ children }) {
...
     }
-  }, [userStates, dispatch])
+  }, [userStates.status, dispatch])
 
   useEffect(() => {
     if (tripsStates.error) {
-      setAlertMessage(tripsStates.error)
-      setAlertOpen(true)
-      setIsLoading(false)
+      enqueueSnackbar(tripsStates.error, { variant: 'error' })
```
Comment (Graywing13) [inline client/src/SessionController.js:129]: nice lol

File: client/src/SessionController.js (hunk 3 above)
Comment (Graywing13) [reply client/src/SessionController.js:167]: o nvm it's by IP. then this is ok

**Review by Graywing13 — APPROVED** [review]
lgtm!!

Verified

- [x] Trip creation decreases count
- [x] Trip update decreases count
- [x] Stage update decreases count
- [x] If the update doesn't go through, it doesn't affect count
- [x] Tracks by IP (diff acc carries over)
- [x] Rename doesnt decrement count
- [x] Can't rename after maxing out
- [x] Alerts stack
- [x] Can't make new edits after maxing out

File: client/src/SessionController.js (hunk 2 above)
Comment (NLmeng) [reply client/src/SessionController.js:120]: it's just being called to fetch the rate limit from the header of the user's request into the userslice, dont need to do it in the actual component

**Pull request merged**